shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/ddl/impl/ShardingCreateViewStatementValidator.java

```diff
@@ -35,11 +45,56 @@ public final class ShardingCreateViewStatementValidator extends ShardingDDLState
     @Override
     public void preValidate(final ShardingRule shardingRule, final SQLStatementContext<CreateViewStatement> sqlStatementContext,
                             final List<Object> parameters, final ShardingSphereDatabase database) {
-        // TODO prevent if has need merge select.
+        Optional<SelectStatement> selectStatement = sqlStatementContext.getSqlStatement().getSelect();
+        if (selectStatement.isPresent()) {
+            TableExtractor extractor = new TableExtractor();
+            extractor.extractTablesFromSelect(selectStatement.get());
+            Collection<SimpleTableSegment> tableSegments = extractor.getRewriteTables();
+            for (SimpleTableSegment each : tableSegments) {
+                String logicTable = each.getTableName().getIdentifier().getValue();
+                if (shardingRule.isShardingTable(logicTable)) {
+                    validateShardingTableAndViewIsBind(shardingRule, sqlStatementContext.getSqlStatement().getView().getTableName().getIdentifier().getValue(), logicTable);
+                }
+            }
+        }
     }
     
     @Override
     public void postValidate(final ShardingRule shardingRule, final SQLStatementContext<CreateViewStatement> sqlStatementContext, final List<Object> parameters,
                              final ShardingSphereDatabase database, final ConfigurationProperties props, final RouteContext routeContext) {
+        Optional<SelectStatement> selectStatement = sqlStatementContext.getSqlStatement().getSelect();
+        selectStatement.ifPresent(statement -> validateNeedsMerge(statement, routeContext));
```

Comment:
Can we rename this method to isContainsNotSupportedViewStatement()? And return boolean type.

Reply:
Thank you. Done.